cli/cmd/set-env-vars.go

```diff
@@ -53,19 +61,50 @@ func (l *SetEnvVarCmd) RunE(_ *cobra.Command, args []string) (err error) {
 }
 
 func (l *SetEnvVarCmd) SetEnvironmentVariables(client Client) (err error) {
-	envVarMap, err := cs.ArgToEnvVarMap(*l.Opts.EnvVar)
-	if err != nil {
-		return fmt.Errorf("failed to parse environment variables: %w", err)
+	finalEnvVarMap := make(map[string]string)
+
+	if l.Opts.EnvFile != nil && *l.Opts.EnvFile != "" {
+		envFile := *l.Opts.EnvFile
+		if _, err := os.Stat(envFile); os.IsNotExist(err) {
+			return fmt.Errorf("env file does not exist: %s", envFile)
+		}
+
+		fileEnvMap, err := godotenv.Read(envFile)
+		if err != nil {
+			return fmt.Errorf("failed to parse env file %s: %w", envFile, err)
+		}
+
+		for key, value := range fileEnvMap {
+			finalEnvVarMap[key] = value
+		}
+	}
+
+	if l.Opts.EnvVar != nil && len(*l.Opts.EnvVar) > 0 {
+		flagEnvVarMap, err := cs.ArgToEnvVarMap(*l.Opts.EnvVar)
+		if err != nil {
+			return fmt.Errorf("failed to parse environment variables from flags: %w", err)
+		}
+
+		for key, value := range flagEnvVarMap {
+			finalEnvVarMap[key] = value
```

Comment:
can we move this to a separate function to keep complexity of this function smaller?